langwatch/src/app/api/llmConfigs/[[...route]]/app.ts

```diff
@@ -0,0 +1,203 @@
+import { Hono } from "hono";
+import { validator as zValidator } from "hono-openapi/zod";
+import { z } from "zod";
+import { describeRoute } from "hono-openapi";
+import { patchZodOpenapi } from "../../../../utils/extend-zod-openapi";
+import { prisma } from "../../../../server/db";
+import type { Project } from "@prisma/client";
+import { LlmConfigRepository } from "../../../../server/repositories/llm-config.repository";
+
+patchZodOpenapi();
+
+// Reuse schema definitions
+const configJsonSchema = z.record(z.any());
+const baseConfigSchema = z.object({
+  name: z.string().min(1, "Name cannot be empty."),
+});
+const baseVersionSchema = z.object({
+  configData: configJsonSchema,
+  schemaVersion: z.string().min(1, "Schema version cannot be empty."),
+  commitMessage: z.string().optional(),
+});
```

Comment:
so, instead of having the typescript definitions elsewhere, and then kinda duplicating it as json schemas, it's much better if we have the definitions to be in json schema directly, and then we can use z.infer to extract the types out of it

initially, we were doing the other way around, types in typescript was the source of truth, and we used ts-to-zod to convert automatically (we still do for most part), but ts-to-zod is extremely limited, throwing many errors, not very reliable.

What is actually 100% reliable and solved any duplication or accidental mismatches is `z.infer`, so then we can have the zod schema be our source of truths, and LlmConfigRepository type would be like:

```typescript
type LlmConfigRepository = z.infer<llmConfigRepositorySchema>
```